enhancements/insights/on-demand-data-gathering.md

```diff
@@ -0,0 +1,340 @@
+}
+
+const (
+	// No data obfuscation
+	NoPolicy DataPolicy = "None"
+	// IP addresses and cluster domain name are obfuscated
+	ObfuscateNetworking DataPolicy = "ObfuscateNetworking"
+	// Data gathering is running
+	Running DataGatherState = "Running"
+	// Data gathering is completed
+	Completed DataGatherState = "Completed"
+	// Data gathering failed
+	Failed DataGatherState = "Failed"
+	// Data gathering is pending
+	Pending DataGatherState = "Pending"
+)
+
+// dataPolicy declares valid data policy types
+// +kubebuilder:validation:Enum="";None;ObfuscateNetworking
+type DataPolicy string
+
+// dataGatherState declares valid gathering state types
+// +kubebuilder:validation:Enum=Running;Completed;Failed;Pending
+type DataGatherState string
+
+type DataGatherStatus struct {
+	// conditions provide details on the status of the gatherer job.
+	// +listType=atomic
+	// +optional
+	Conditions []metav1.Condition `json:"conditions"`
+	// dataGatherState reflects the current state of the data gathering process.
+	// +optional
+	State DataGatherState `json:"dataGatherState,omitempty"`
+	// gatherers is a list of active gatherers (and their statuses) in the last gathering.
+	// +listType=atomic
+	// +optional
+	Gatherers []GathererStatus `json:"gatherers,omitempty"`
+	// startTime is the time when Insights data gathering started.
+	// +optional
+	StartTime metav1.Time `json:"startTime,omitempty"`
+	// finishTime is the time when Insights data gathering finished.
+	// +optional
+	FinishTime metav1.Time `json:"finishTime,omitempty"`
+	// relatedObjects is a list of resources which are useful when debugging or inspecting the data
+	// gathering Pod
+	// +optional
+	RelatedObjects []ObjectReference `json:"relatedObjects,omitempty"`
+	// insightsRequestID is an Insights request ID to track the status of the 
+	// Insights analysis (in console.redhat.com processing pipeline) for the corresponding Insights data archive. 
+	// +optional
+	InsightsRequestID string `json:"insightsRequestID,omitempty"`
+}
+
+// gathererStatus represents information about a particular
+// data gatherer.
+type GathererStatus struct {
+	// conditions provide details on the status of each gatherer.
+	// +listType=atomic
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	Conditions []metav1.Condition `json:"conditions"`
+	// name is the name of the gatherer.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:MinLength=5
+	Name string `json:"name"`
+	// lastGatherDuration represents the time spent gathering.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Pattern="^([1-9][0-9]*(\\.[0-9]+)?(ns|us|µs|ms|s|m|h))+$"
+	LastGatherDuration metav1.Duration `json:"lastGatherDuration"`
+}
+
+// ObjectReference contains enough information to let you inspect or modify the referred object.
+type ObjectReference struct {
```

Comment:
missing namespace.  See the clusteroperatorstatus for details.  Can be worked out in the api PR.

Reply:
The namespace is supposed to be every time the same. https://github.com/openshift/api/pull/1365#discussion_r1043653004. I can mentioned it somewhere as well.

Reply:
I moved the resource to a cluster scoped one and added the namespace attribute here.